### PR TITLE
Making the pre-check even more complete

### PR DESCRIPTION
**Pull request by danquah**
We now check all variables before accessing them.

OS2-57

### CODE REVIEW

File: web/templates/js/dingEventsSlide.js [hunk 1]
```diff
@@ -155,31 +155,34 @@ if (!window.slideFunctions['ding-events']) {
...
-
-      // Check that external_data exists, if not stay on for one duration and
-      // continue to next slide.
-      if (!slide.external_data || !slide.external_data.events || slide.external_data.events.length <= 0) {
-        // Go straight to the next slide if we don't have any data. For now this
-        // simply assumes that we have a "next" to go to, if not, we're going
+      // Experience has shown that we can't be certain that all our data is
+      // present, so we'll have to be careful verify presence before accessing
+      // anything.
+      if (!slide.options || !slide.external_data || !slide.external_data.events || slide.external_data.events.length <= 0 || !slide.event_settings) {
```
Comment (kasperg) [inline web/templates/js/dingEventsSlide.js:161]: Why do we need the check for `slide.options` and `slide.options` here? Neither seems to be referenced within the scope of this `if` statement ending on L180.

However `slide.options` *is* referenced later on on L184.

File: web/templates/js/dingEventsSlide.js [hunk 2]
```diff
@@ -155,31 +155,34 @@ if (!window.slideFunctions['ding-events']) {
...
+
+        // We tried the skip, did not work, continue to next slide.
         region.itkLog.info("No data for slide, skipping");
 
         region.nextSlide();
         return;
       }
-
       // Reset throttle in case we where successful.
       slide.loop_throttle = false;
```
Comment (kasperg) [inline web/templates/js/dingEventsSlide.js:182]: Given the changes on L169+L171 should this be `slide.loop_skip`?

Comment (danquah) [reply web/templates/js/dingEventsSlide.js:182]: argh, good catch.

File: web/templates/js/dingEventsSlide.js (hunk 1 above)
Comment (danquah) [reply web/templates/js/dingEventsSlide.js:161]: The theory is: If we're missing slide.options something is wrong and we should bail outwhich is why we check for !slide.options and return if it resolves as true.

Comment (kasperg) [reply web/templates/js/dingEventsSlide.js:161]: Ah - right. This is the other way around compared to how I immediately perceived it. This is checks for missing data and immediate ´handling of that situation - not a guard against missing data before executing expected business logic.

**Pull request merged**